src/StripeTests/BaseStripeTest.cs

```diff
@@ -0,0 +1,158 @@
+namespace StripeTests
+{
+    using System;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+
+    using Stripe;
+
+    public class BaseStripeTest : IDisposable
+    {
+        /// <value>Minimum required version of stripe-mock</value>
+        /// <remarks>
+        /// If you bump this, don't forget to bump `STRIPE_MOCK_VERSION` in appveyor.yml as well.
+        /// </remarks>
+        private const string MockMinimumVersion = "0.19.0";
+
+        private static string port;
+
+        private string originalApiBase;
+        private string originalApiKey;
+        private string originalUploadBase;
+
+        public BaseStripeTest()
+        {
+            if (string.IsNullOrEmpty(port))
+            {
+                CheckStripeMock();
+            }
+
+            this.SetUpStripeMock();
+        }
+
+        public void Dispose()
+        {
+            this.TearDownStripeMock();
```

Comment:
I added extra tests and this line causes one of them to fail. It works fine without it which seems to point to some kind of parallelism issue.

Reply:
Ah right, I guess that one test could restore the original state while another test is ongoing.

Honestly, the right answer here is to disable parallel test execution as long as we have to rely on global state. It shouldn't be too much of an issue since stripe-mock tests are fast anyway.

Reply:
But if the test suites runs separately (like as 3 separate projects they must right?) then that new one does not need the hack since it would only ever use stripe-mock right? This was only needed before you decided to add a third project I think.

Reply:
Yes, the different test projects are run separately. Here's what's currently happening in the new project:
- T1: call `SetUpStripeMock()`
- T1: run a test
- T2: call `SetUpStripeMock()` (effectively a no-op since the values are the same)
- T1: call `TearDownStripeMock()`
- T2: run a test --> fails because T1 restored the original state values and so T2's requests are sent to the live API servers instead of stripe-mock

We could simply set the global state once then never clean up and that would probably work, but it seems less hacky to me to disable parallel execution until we no longer have to rely on global state (i.e. until we can provide `ApiBase` / `UploadBase` on a per-request or per-service basis).

Reply:
It just seems cleaner to me to have isolated tests that don't rely on global state.

That said, it looks like stripe-go does exactly this (i.e. sets the global configuration once and never restore it afterwards) so I've updated the PR to do the same thing.